invenio_i18n/cli.py

```diff
@@ -10,228 +10,782 @@
+        secho(f"Package {package} not found")
+        return
+
+    po_path = next(
+        (
+            path
+            for loc, path in find_js_po_files(package_root, package)
+            if loc == locale
+        ),
+        None,
+    )
+    if not po_path:
+        secho(f"No JavaScript PO file for {package} in {locale}")
+        secho(
+            "Hint: JavaScript PO files are typically in assets/semantic-ui/translations/<package>/messages/<locale>/messages.po",
+        )
+        return
+
+    updated, created_new, count, error = update_po_file(po_path, msgid, msgstr, prefix)
+    if error:
+        secho(f"Error: {error}")
+        return
+    if not updated:
+        return
+    target_name = f"{package}/{locale} (JS)"
+    if prefix:
+        secho(
+            f"Updated {count} translation(s) matching '{msgid}' in {target_name}",
+            fg="green",
+        )
+    elif created_new:
+        secho(f"Created '{msgid}' in {target_name}", fg="green")
+    else:
+        secho(f"Updated '{msgid}' in {target_name}", fg="green")
+    try:
+        po_file = polib.pofile(str(po_path))
+        json_data = po_to_i18next_json(po_file, package)
 
+        json_path = po_path.parent / "translations.json"
+        json_path.parent.mkdir(parents=True, exist_ok=True)
 
-    Distribution of translation
-    ---------------------------
-    This CLI command processes unified per‑language JSON files in a given input path.  The command extracts translations that belong to the target package, discovers asset root paths of packages through the ``invenio_assets.webpack``
-    entry‑point group and writes it to the package’s  translation folder in react-i18next format here.
+        with json_path.open("w", encoding="utf-8") as fp:
+            dump(json_data, fp, indent=2, ensure_ascii=False)
 
-    For example, for locale ``tr`` the extracted fragment for
-    ``invenio_communities`` is written to:
+        secho(f"Converted to JSON: {json_path}")
+    except (OSError, IOError, FileNotFoundError, PermissionError, ValueError) as e:
+        secho(f"Warning: Failed to convert to JSON: {e}")
 
-    ``<site‑packages>/invenio-communities/assets/semantic-ui/translations/invenio_communities/messages/tr/translations.json``
 
-    Missing directories and files will be created automatically if not exist.
+@js_translation.command("build")
+@option(
+    "--packages",
+    "-p",
+    multiple=True,
+    callback=_convert_to_list,
+    help="Packages to build. Can be specified multiple times.",
+)
+@option(
+    "--all-packages",
+    "--global",
+    is_flag=True,
+    help="Build JavaScript translations for all invenio_* packages",
+)
+@option(
+    "--output-directory",
+    "-o",
+    type=ClickPath(
+        exists=False, file_okay=False, dir_okay=True, writable=True, path_type=Path
+    ),
+    default=Path.cwd() / "js-translations",
+    help="Directory for temporary JSON files. Default: ./js-translations",
+)
+def build_js_translations(
```

Comment:
nit: this command is also to long, please split the functionality into functions which should be moved to somewhere else.